### DISCUSSION TITLE
How to Add the Empty Spaces indentation.

### DISCUSSION THREAD

**Discussion by SohamTilekar** [squidfunk/mkdocs-material#6915] [Q&A]
How can we add the empty spaces indentation to the app without converting it into the code block.
E.G: -
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/2b1717b9-58e8-4c92-99de-cb60921f6cb1)
How Do I Indent like that.

Currently it Convert it into the code block.
E.G: -
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/476342a2-55e8-446f-accf-78f987ecf204)

**Comment by alexvoss** [comment]
If you need to document Python code then you should probably have a look at [mkdocstrings](https://mkdocstrings.github.io/) if you have not done that yet.

Custom uses of indentation  will be a bit difficult to implement. It also is a bit dangerous in the sense  that it is difficult to consistently get it right. If mkdocstrings does not work for you then I would recommend using [attribute lists](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown/#attribute-lists) to specify what a line or block is, then use CSS to produce the indentation.

**Reply by kamilkrzyskow** [reply]
They can also use a HTML div tag: https://squidfunk.github.io/mkdocs-material/reference/annotations/#in-everything-else 
to select a whole range of paragraphs to be indented with a specified CSS class.

**Reply by SohamTilekar** [reply]
I don't know css can you give me the basic tample.

**Reply by SohamTilekar** [reply]
Is there any way to use this type of syntax which is provided in the mkdocs documentation. But how can we use it with the material theme.
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/0751630a-3d59-4d53-a154-b329c16fab26)

**Reply by alexvoss** [reply]
The example you posted is a [Jinja2](https://palletsprojects.com/p/jinja/) template. Jinja2 is the templating engine that MkDocs uses to produce the HTML output. Material at its heart consists of [a number of such templates](https://github.com/squidfunk/mkdocs-material/tree/master/src/templates) plus a CSS stylesheet that defines how the HTML is rendered by the browser. 
I do not think you need to mess with Jinja2 but can use the approaches @kamilkrzyskow and I suggested for producing HTML in the required form.

Some [knowledge of CSS](https://developer.mozilla.org/en-US/docs/Learn/CSS/First_steps/What_is_CSS) to produce the necessary indentation plus any other styling you want will be needed, I am afraid. If you attach a CSS class `indent` to a Markdown block, for example, you should be able to have it indented using the [margin attributes](https://developer.mozilla.org/en-US/docs/Web/CSS/margin).

*for completeness:* if you want to write your own Jinja2 templates then you can do this by declaring a `custom_dir` in your `mkdocs.yml`. You should read the [customization guide](https://squidfunk.github.io/mkdocs-material/customization) in any case.

**Comment by SohamTilekar** [comment]
Thanks for your Solutions I Found the Way For custom indenting.
E.G: -
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/33f7a881-e2db-4910-a42d-cb49c7318994)
output: -
![image](https://github.com/squidfunk/mkdocs-material/assets/152298897/c7bac17d-1766-4b44-bf9f-48466e2d7266)

I don't Know what it is called which extension is used to provide this result.
